src/middlewares/patient.js

```diff
@@ -3,18 +3,26 @@
 
     var keyFor = {
         "Observation": "subject",
-        "MedicationPrescription": "patient"
+        "MedicationPrescription": "patient",
+        "Encounter": "patient"
```

Comment:
All I'm saying is that *technically* providing a "patient=123" parameter on
arbitrary search operations should be okay. Any time there's a patient
associated with the resource, the search param will be correctly called
"patient". And any time there isn't, it should be ignored (at least, that's
FHIR's intended behavior for unrecognized query parameters). Of course,
having a list of resources would be nice, so we don't say "GET
/ValueSet?patient=123" -- but then again, a developer using out client
shouldn't be searching for value sets with the patient-context-embedded
object; the should use the patient-context-free object for that kind of
search.

On Thu, Sep 3, 2015 at 11:25 AM, nschwertner <notifications@github.com>
wrote:

> In src/middlewares/patient.js
> <https://github.com/FHIR/fhir.js/pull/50#discussion_r38658563>:
>
> > @@ -3,18 +3,26 @@
> >
> >      var keyFor = {
> >          "Observation": "subject",
> > -        "MedicationPrescription": "patient"
> > +        "MedicationPrescription": "patient",
> > +        "Encounter": "patient"
>
> Josh, sure, we can expose two instances of the client as you outlined with
> and without the patient context middleware enabled. Furthermore, the client
> can decide which of the two client to instantiate and expose based on the
> launch conditions (if there is a patient context). This way, an app which
> expects to work in patient context will not produce incorrect results by
> accident.
>
> Still, this does not resolve the need of the patient context middleware to
> have a list of the resources for which it should apply the patient context
> (unless there is something that I don't yet understand here).
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/FHIR/fhir.js/pull/50/files#r38658563>.
>